sound/soc/sof/pcm.c

```diff
@@ -338,8 +339,10 @@ static snd_pcm_uframes_t sof_pcm_pointer(struct snd_pcm_substream *substream)
 
 	dev_dbg(sdev->dev, "PCM: stream %d dir %d DMA position %lu DAI position %lu\n",
 		spcm->pcm.pcm_id, substream->stream, host, dai);
-
 	return host;
+#else
+	return	snd_sof_pcm_platform_pointer(sdev, substream);
+#endif
```

Comment:
Ah so this one is very interesting. Why not rely on DPIB all the time? Is there a reason why we need to reason from the DSP?
It really looks like either the DSP provides the information, or the DMA does it through the pointer information. This should be a platform property, not hardcoded with a define.

Reply:
that's true, the new update is that the DPIB works for DSP I2S also on my local platform, I will run more test and if no regression then we can switch to use DPIB for all mode(I still want to keep IPC position update as another choice though).